### ISSUE TITLE
File copy does not work correctly

### ISSUE THREAD

**Issue by DenisKOMAROV** [superannotateai/annotateonline-input-converters#6]
Hello! When I run this command:
```
python3 sa_to_coco.py -is ./dataset/steel_sheet_coil_markup/ -sr 90 -ptype pixel -t instance_segmentation -dn dataset_name -od ./dataset/instance_segmentation/ -cp True
```
I got such error:
```
WARNING:root:Source image files will be copied to output_dir/test and output_dir/train folders
Traceback (most recent call last):
  File "/home/deniskomarov/IDE/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 1434, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/deniskomarov/IDE/pycharm/plugins/python-ce/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/deniskomarov/PROJECTS/Python/annotateonline-input-converters/sa_to_coco.py", line 291, in <module>
    move_files(train_set=train_set, test_set=test_set, src='./dataset/steel_sheet_coil_markup/', cp=True)
  File "/home/deniskomarov/PROJECTS/Python/annotateonline-input-converters/sa_to_coco.py", line 184, in move_files
    move_fn(i, os.path.join(test_path, i.split('/')[-1]))
  File "/usr/lib/python3.6/shutil.py", line 245, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.6/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
IsADirectoryError: [Errno 21] Is a directory: './dataset/steel_sheet_coil_markup/classes'
```
I've been debugging the code and got this:
```
test_set = {ndarray: (1, 3)} [['./dataset/steel_sheet_coil_markup/classes',  './dataset/steel_sheet_coil_markup/classes___pixel.json',  './dataset/steel_sheet_coil_markup/classes___save.png']]
```
It seems like test_set should be files only, because there is no classes___save.png or classes___pixel.json.
Maybe using os.listdir would help. Or somehow changing this code.
https://github.com/superannotateai/annotateonline-input-converters/blob/41dca484582d24b9ea26e3bfe9d6af46dba5e892/sa_to_coco.py#L180-L183

**Comment by VavoTK** [comment]
Could you pull once again and try it? is it still failing?

**Comment by DenisKOMAROV** [comment]
@VavoTK Thanks for the quick reply
Yes, I have pulled the code and the issue stays:
```
(venv_sa_conv) ...:~/PROJECTS/Python/annotateonline-input-converters$ python3 sa_to_coco.py -is ./dataset/steel_sheet_coil_markup/ -sr 80 -ptype pixel -t instance_segmentation -dn dataset_name -od ./dataset/instance_segmentation3 -cp True
WARNING:root:Source image files will be copied to output_dir/test and output_dir/train folders
Traceback (most recent call last):
  File "sa_to_coco.py", line 293, in <module>
    move_files(train_set=train_set, test_set=test_set, src='./dataset/steel_sheet_coil_markup/', cp=True)
  File "sa_to_coco.py", line 186, in move_files
    move_fn(i, destination)
  File "/usr/lib/python3.6/shutil.py", line 245, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.6/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
IsADirectoryError: [Errno 21] Is a directory: './dataset/steel_sheet_coil_markup/classes'
```
I'm not sure, but I guess current code takes all the names in src directory (while using shutil.copy()) and first object that it encounters is the "classses/" directory.
I'm figuring out how the code works atm, but a possible solution is using os.listdir or os.walk

**Comment by VavoTK** [comment]
@DenisKOMAROV 
Your diagnosis is right based on the error message, but I cannot reproduce the issue on my side. The code works pretty simple I collect all the images files into a variable called `orig_images` on lines 125 and 126 of the `sa_to_coco.py`. And the line 
`orig_images = [x for x in orig_images if not os.path.isdir(x) and  '___' not in x.split('.')[-2] and  "mapper.json" not in x]` should exclude all non-necessary files and folders from the list of images. 

Can I get a small sample of your ZIP file on which you are running the conversion? 

A small (and very ugly ) fix that I can suggest if everything else fails is to wrap the actual moving of the files in a `try ... except` block like this 
```
if train_set is not None:
      for tup in train_set:
          for i in tup:
              try:
                  move_fn(i, os.path.join(train_path, i.split('/')[-1]))
               except Exception as e: 
                   pass
 if test_set is not None:
      for tup in test_set:
          for i in tup:
              try:
                  move_fn(i, os.path.join(test_path, i.split('/')[-1])) 
              except Exception as e:
                  pass
```

**Comment by DenisKOMAROV** [comment]
@VavoTK Hello! Thank you for the provided solution and sorry for not replying for so long.

I was trying to reproduce the same case using the test cats&dogs dataset that is included in repository (I didn't notice it at the start of my experiments). The error is gone now. I checked all the differences between my and the test dataset and noticed that my dataset does not contain the original images, for some reason I expected to see the marked images __save.png in the output folder. This is my fault.
Your code works just fine. I'm closing this issue. Sorry for wasting your time.

TLDR: When converting images from annotate online to COCO,
don't forget to copy original images into the folder that you get from annotate online
